reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp

```diff
@@ -356,6 +364,26 @@ std::optional<double> EarVstExportSources::getValueFor(std::shared_ptr<admplug::
     return std::optional<double>();
 }
 
+bool EarVstExportSources::getEnvelopeBypassed(TrackEnvelope* env, ReaperAPI const& api)
+{
+    bool envBypassed = false;
+    char chunk[1024]; // A new envelope should only be about 100 bytes
+    bool getRes = api.GetEnvelopeStateChunk(env, chunk, 1024, false);
+    if (getRes) {
+        std::istringstream chunkSs(chunk);
+        std::string line;
+        while (std::getline(chunkSs, line)) {
+            auto activePos = line.rfind("ACT ", 0);
+            if (activePos != std::string::npos) {
+                auto active = static_cast<int>(line.at(activePos + 4)) - 48;
```

Comment:
Potentially two issues here. 
First, in the (unlikely) event the chunk got truncated so the returned string ended with "ACT " (see link in comment about size assumption) I think the `rfind()` would succeed but the `.at()` would throw an out of range exception as std::strings don't store the terminating char.
Second, is there a reason for the cast / ascii arithmetic or could you just do 
```c++
envBypassed = line.at(activePos + 4) == '0';
```